xarray/test/test_dataset.py

```diff
@@ -1868,8 +1868,9 @@ def test_fillna(self):
         expected = Dataset({'a': ('x', [-1, 1, -1, 3]), 'b': np.nan})
         self.assertDatasetIdentical(expected, actual)
 
-        with self.assertRaisesRegexp(ValueError, 'no overlapping'):
-            ds.fillna({'x': 0})
+        actual=ds.fillna({'x': 0})
+        expected=ds
+        self.assertDatasetEqual(actual, expected)
```

Comment:
I think these may now be *too* liberal. Here, there is no variable called `x`. I think this _should_ raise - it seems fairly different to multiply empty arrays vs supplying variables that don't exist.

I'll ponder further, let me know if you have any thoughts on this in the meantime